Wangscape/AlphaCalculator.cpp

```diff
@@ -0,0 +1,85 @@
+#include "AlphaCalculator.h"
+#include <algorithm>
+#include <numeric>
+
+
+
+AlphaCalculator::AlphaCalculator()
+{
+}
+
+
+AlphaCalculator::~AlphaCalculator()
+{
+}
+
+void AlphaCalculator::calculatePixelAlphaLinear(WeightVector& weights, AlphaVector& alphas)
+{
+    int err = 255;
+    float total_weight = 0.f;
+    for (auto& weight : weights)
+    {
+        weight = std::max(0.f, weight);
```

Comment:
I accidentally passed negative weights to this function while developing this process, and it's inevitable that end-users will too, so the weight-to alpha conversion needs to be able to cope with negative inputs without crashing.

But you're right that applying a lower bound should not be this function's job - that should be split into a separate function.